test/scripts/products/h24-verify-selfmanaged-apicast-and-custom-policy/h24-verify-selfmanaged-apicast-and-custom-policy.go

```diff
@@ -0,0 +1,576 @@
+		if err != nil {
+			log.Error("Error get Customer Amin Password. Exit...", err)
+			return
+		}
+		err = createCustomerUsers(customerAdminPassword)
+		if err != nil {
+			log.Error("Error create Customer Users. Exit...", err)
+			return
+		}
+	} else {
+		log.Info("Skipped creation of testing-idp. To enable it - update test.sh - set -create-testing-idp=\"yes\" ")
+	}
+
+	// Customer login
+	if useCustomerAdminUser {
+		err = customerLogin()
+		if err != nil {
+			log.Error("Error in customer login. Exit...", err)
+			return
+		}
+	}
+
+	// Import APIcast Image
+	err = importApicastImage()
+	if err != nil {
+		log.Error("Error importApicastImage(), Exit...", err)
+		return
+	}
+
+	// Create an adminportal-credentials secret
+	threeScaleAdminPortal, err := getThreeScaleAdminPortal(ctx, client)
+	if err != nil {
+		log.Error("Error get 3scale admin portal. Exit...", err)
+		return
+	}
+	err = createAdminPortalCredentialsSecret(ctx, client, token3scale, threeScaleAdminPortal)
+	if err != nil {
+		log.Error("Error create adminportal-credentials secret. Exit...", err)
+		return
+	}
+	time.Sleep(5 * time.Second)
+
+	// Use self-managed APIcast instead of the builded one for API
+	userKey, err := promoteSelfManagedAPIcast(ctx, client, promoteManually)
+	if err != nil {
+		log.Error("Error promote Self-managed APIcast. Exit...", err)
+		return
+	}
+
+	// Install 3scale APIcast gateway operator
+	err = installThreeScaleApicastGatewayOperator(ctx, client)
+	if err != nil {
+		log.Error("Error install 3scale APIcast gateway Operator. Exit...", err)
+		return
+	}
+	time.Sleep(60 * time.Second) //TODO- recheck if sleep needed
+
+	// Create a self-managed APIcast
+	err = createSelfManagedApicast(ctx, client)
+	if err != nil {
+		log.Error("Error create self-managed APIcast. Exit...", err)
+		return
+	}
+
+	// Create a route for the self-managed APIcast
+	routeHost, err := createApicastRoute(ctx, client, threeScaleAdminPortal)
+	if err != nil {
+		log.Error("Error create route for self-managed APIcast. Exit...", err)
+		return
+	}
+
+	// Validation of the Deployment
+	res, err := validateDeployment(userKey, routeHost)
+	if err != nil {
+		log.Error("Validation of Self-managed APIcast API gateway Deployed - Failed", err)
+	} else {
+		if res {
+			log.Info("Validation of Self-managed APIcast API gateway Deployment - Succeeded")
+		} else {
+			log.Error("Self-managed APIcast API gateway Deployment issue", nil)
+		}
+	}
+
+	log.Info("Self-managed APIcast API gateway - Deployment script Completed")
+
+} //end of main()
+
+// Create customer users in dedicated-admins group
+func createCustomerUsers(customerAdminPassword string) error {
+	log.Info("Creating customer users in dedicated-admins group")
+	command := "PASSWORD=" + customerAdminPassword + " DEDICATED_ADMIN_PASSWORD=" + customerAdminPassword + " ../../../../scripts/setup-sso-idp.sh"
+	err := run_shell_command(command)
```

Comment:
Good question. I think the follow-up could be to run this script automatically (if `create-testing-idp == true`) and get a password value from a parameter, otherwise use a default password.

Reply:
The plan - to integrate the script into Jenkins Pipeline. All parameters will be entered from Pipeline, and script will be automated, but it will be another, new task.  At the same we will have option for interactive working with script. Script will be tailored.